src/ivc/cyclefold/sfc/mod.rs

```diff
@@ -32,9 +36,28 @@ pub struct StepFoldingCircuit<
 
 impl<const ARITY: usize, C: CurveAffine, SC: StepCircuit<ARITY, C::ScalarExt>>
     StepFoldingCircuit<'_, ARITY, C, SC>
+where
+    C::ScalarExt: PrimeFieldBits + FromUniformBytes<64>,
 {
-    pub fn instances(&self) -> Vec<Vec<C::ScalarExt>> {
-        todo!()
+    /// For the initial iteration, we will give the same accumulators that we take from the input
+    pub fn initial_instances(&self) -> Vec<Vec<C::ScalarExt>> {
+        let marker = cyclefold::ro()
+            .absorb(&self.input)
+            .output(NonZeroUsize::new(<C::ScalarExt as PrimeField>::NUM_BITS as usize).unwrap());
+
+        let mut instances = self.sc.instances();
+        instances.insert(0, vec![marker, marker]);
+        instances
+    }
+
+    pub fn instances(&self, expected_out: C::ScalarExt) -> Vec<Vec<C::ScalarExt>> {
+        let input_marker = cyclefold::ro()
+            .absorb(&self.input)
+            .output(NonZeroUsize::new(<C::ScalarExt as PrimeField>::NUM_BITS as usize).unwrap());
+
+        let mut instances = self.sc.instances();
+        instances.insert(0, vec![input_marker, expected_out]);
```

Comment:
what are the meaning of input_marker and expected_out?

Reply:
This is consistency markers, same as we had in ivc::sanrgia (X0, X1)

`Expected out` this is a hash of output data

Reply:
in `ivc::sangaria`, `X0, X1` are hashed IO of self and paired circuits. 
Now in cyclefold, since we don't have cycles of curves. I suppose we will only have one marker `X0` that is the hash of "self circuit". 

To be more specific, we have something like this:
`X0=hash(i,z0,zi, pp, U, u, U*, u*, aux)`

Reply:
Then you can reveal the idea of what and how we compare it with. In fact, at step N we have only N-1 input, which will not help us in any way

Here we have calculated the instance of the current input and let's say it was substituted.

How at the verification stage, not having all the steps, should we verify that the folded marker is corret?

Reply:
Yes, we can check input consistency as follows:
```
// input checking:
u_i.x=hash(i,z0,zi, pp, U_i, U*_i, aux)


// instance of the current step
// this will be u_{i+1}.x
X0 = hash(i+1,z0,z_{i+1}, U_{i+1}, U*_{i+1}, aux)
```

`

Reply:
But if we have only one consistency marker in instances[0][0], then how we can in same time:
- have something to check input
- have something to check input at next step
using only one hash?

In you last comment we have only one hash with `input`.

Reply:
It's just the last thing I have left to sfc.

Here I have the input and output ready.

I need to:
- Check for acceptability with the previous step
- Prepare data for acceptance with the next step

How to do this through a single marker?

Reply:
Ok, we can access to prev-marker via self_trace and current marker via cs. Got it

Reply:
Fixed in #399

Reply:
will mark this as resolved